src/gas/mod.rs

```diff
@@ -480,20 +456,30 @@ pub fn inject_gas_counter(module: elements::Module, rules: &rules::Set)
 				for ref mut segment in elements_section.entries_mut() {
 					// update all indirect call addresses initial values
 					for func_index in segment.members_mut() {
-						if *func_index >= gas_func { *func_index += 1}
+						if *func_index >= out_of_gas_callback { *func_index += 1}
 					}
 				}
 			},
 			&mut elements::Section::Start(ref mut start_idx) => {
-				if *start_idx >= gas_func { *start_idx += 1}
+				if *start_idx >= out_of_gas_callback { *start_idx += 1}
 			},
 			_ => { }
 		}
 	}
 
 	if error { return Err(module); }
 
-	if need_grow_counter { Ok(add_grow_counter(module, rules, gas_func)) } else { Ok(module) }
+ 	Ok(module)
+}
+
+pub fn set_total_gas(module: elements::Module, total_gas: i32) -> Result<elements::Module, ()> {
```

Comment:
Hm, could you explain why we need this? It seems to me that this fixes the gas amount during contract instrumentation time. I would expect, instead, to set the available gas at runtime, so that we don't have to re-instrument the code at every execution.

Reply:
Yes available gas should be set at runtime. I keep this intended to make the gas meter library self-complete: need a utility to set gas other than zero.

> we don't have to re-instrument the code at every execution.

Good point. re-instrument is expensive, should set it by wasmer Memory/Module